backend/pkg/repository/groupRepository.go

```diff
@@ -89,4 +89,18 @@ func (r *GroupRepository) DeleteGroup(id int) error {
 	query := `DELETE FROM groups WHERE id = ?`
 	_, err := r.db.Exec(query, id)
 	return err
-}
\ No newline at end of file
+}
+
+// RemoveGroupMembers removes all group members of a specific group.
+func (r *GroupMemberRepository) RemoveGroupMembers(groupID int) error {
+	query := `DELETE FROM group_members WHERE group_id = ?`
+	_, err := r.db.Exec(query, groupID)
+	return err
+}
+
+// LogGroupDeletion logs the deletion of a group.
+func (r *GroupRepository) LogGroupDeletion(groupID int) error {
```

Comment:
the idea of it is correct, but the groups table currently doesn't have the field 'deleted'